### PR TITLE
TLS 1.3: Add specific field to store the selected key exchange mode

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
PR to help finalize #6106

### CODE REVIEW

File: library/ssl_misc.h [hunk 1]
```diff
@@ -611,7 +611,8 @@ struct mbedtls_ssl_handshake_params
      * Handshake specific crypto variables
      */
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3)
-    int tls13_kex_modes; /*!< key exchange modes for TLS 1.3 */
+    uint8_t key_exchange_mode; /*!< Selected key exchange mode */
```
Comment (yuhaoth) [inline library/ssl_misc.h:614]:
```suggestion
    uint8_t key_exchange_mode; /*!< Selected negotiation key exchange mode */
```

File: library/ssl_misc.h [hunk 2]
```diff
@@ -611,7 +611,8 @@ struct mbedtls_ssl_handshake_params
...
+    uint8_t tls13_kex_modes;   /*!< key exchange modes for TLS 1.3 */
```
Comment (yuhaoth) [inline library/ssl_misc.h:615]:
```suggestion
    uint8_t tls13_kex_modes;   /*!< received key exchange modes from client */
```
Should we guards with `defined(MBEDTLS_SSL_SRV_C) && defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)` ?

**Review by yuhaoth — CHANGES_REQUESTED** [review]
Almost good to me. Two minor comments

File: library/ssl_misc.h (hunk 1 above)
Comment (ronald-cron-arm) [reply library/ssl_misc.h:614]: This makes it more confusing to me. I hesitated between "Negotiated" and "Selected" but opted for "Selected" as it is not really negotiated in the sense that the server does not send its key exchange mode choice to the client. It is deduced from the content of the ClientHello/ServerHello and the endpoint configuration. All together I prefer to keep it as it is.

Comment (yuhaoth) [reply library/ssl_misc.h:614]: Got it.

**Review by xkqian — APPROVED** [review]
LGTM

**Comment by yuhaoth** [comment]
Travis CI got a python install fail. I think it can be ignore.

**Pull request merged**